### PR TITLE
Make compactor podManagementPolicy configurable

### PR DESCRIPTION
**Pull request by blazarecki**
#### What this PR does

This PR allow to configure the compactor `podManagementPolicy`

#### Which issue(s) this PR fixes or relates to

None

#### Checklist

- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by krajorama** [comment]
Hi, have you tried upgrade from previous version? I think changing the PodManagementPolicy is prohibited during upgrade, so the default value cannot be "Parallel". 

Also , please run `make check-helm-tests`  to update the golden test records in the PR.

**Comment by krajorama** [comment]
Please take a look at #5757 that did the same for store gateways.

**Comment by blazarecki** [comment]
Thanks for the feedback, I updated the PR.

File: operations/helm/tests/enterprise-https-values-generated/mimir-distributed/templates/compactor/compactor-statefulset.yaml [hunk 1]
```diff
@@ -82,7 +83,7 @@ spec:
         - name: runtime-config
           configMap:
             name: enterprise-https-values-mimir-runtime
-        
+
```
Comment (krajorama) [inline operations/helm/tests/enterprise-https-values-generated/mimir-distributed/templates/compactor/compactor-statefulset.yaml:86]:
```suggestion
```

File: operations/helm/tests/enterprise-https-values-generated/mimir-distributed/templates/compactor/compactor-statefulset.yaml [hunk 2]
```diff
@@ -96,7 +97,7 @@ spec:
             - "-config.expand-env=true"
             - "-config.file=/etc/mimir/mimir.yaml"
           volumeMounts:
-            
+
```
Comment (krajorama) [inline operations/helm/tests/enterprise-https-values-generated/mimir-distributed/templates/compactor/compactor-statefulset.yaml:100]:
```suggestion
```

File: operations/helm/tests/enterprise-https-values-generated/mimir-distributed/templates/compactor/compactor-statefulset.yaml [hunk 3]
```diff
@@ -82,7 +83,6 @@ spec:
         - name: runtime-config
           configMap:
             name: enterprise-https-values-mimir-runtime
-
```
Comment (krajorama) [inline operations/helm/tests/enterprise-https-values-generated/mimir-distributed/templates/compactor/compactor-statefulset.yaml:85]: Linter fails on this and when I run `make check-helm-tests` I don't get this change. Same for line 99 bellow.

Comment (blazarecki) [reply operations/helm/tests/enterprise-https-values-generated/mimir-distributed/templates/compactor/compactor-statefulset.yaml]: Thanks for the feedback, I just run the `make check-helm-tests`.

**Review by krajorama — APPROVED** [review]

**Pull request merged**